### PR TITLE
fix: use file picker component directly

### PR DESCRIPTION
**Pull request by elzody**
* Resolves: #3957 
* Target version: main

### Summary
The previous implementation stopped working because the file picker's `container` prop expects a selector, but an element was passed instead. This PR fixes that, and uses the file picker component directly.

### Checklist

- [x] Code is properly formatted
- [x] Sign-off message is added to all commits
- [x] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: src/view/DocumentTargetPicker.vue [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 <template>
 	<div v-if="filePath === null" class="office-target-picker">
-		<div ref="picker" class="reference-file-picker" />
+		<div class="reference-file-picker" />
```
Comment (juliushaertl) [inline src/view/DocumentTargetPicker.vue:8]: We could actually use the vue component directly https://github.com/nextcloud-libraries/nextcloud-dialogs?tab=readme-ov-file#use-the-vue-component-directly

That way we would not have nested modals

Comment (elzody) [reply src/view/DocumentTargetPicker.vue]: Indeed, it has been improved, however the FilePickerVue component itself is actually wrapped in an NcDialog
https://github.com/nextcloud-libraries/nextcloud-dialogs/blob/016d046e79b949838bb78d01dca9bcb1e1db286a/lib/components/FilePicker/FilePicker.vue#L6

Comment (juliushaertl) [reply src/view/DocumentTargetPicker.vue]: All right, then fine either way 😄

**Comment by juliushaertl** [comment]
Could you also add a cypress test for that?

**Comment by juliusknorr** [comment]
@elzody took me quite a bit but found why this failed. On your local setup it likely worked because you had text enabled which dispatches an RenderReferenceEvent. This event is listened for to [provide the list of providers](https://github.com/nextcloud/server/blob/dae7c159f728a90ffa53247d6e033abdae5d2bd6/lib/private/Collaboration/Reference/RenderReferenceEventListener.php#L41). I left my commits with debugging attempts for you to follow along for now. Feel free to clean them up and get this merged.

After seeing that the initial state was empty if checked where in the server that was set. Then I locally set a breakpoint to see which event dispatching actually triggers the listener on my local setup. That was text, so disabling text showed the same beahavior as CI when running the cypress test.

**Comment by elzody** [comment]
@juliusknorr Thanks for the help, seems like it required some really deep debugging. I think I managed to clean up the nonsense debug commits, and it seems like everything passes now!

**Review by juliusknorr — APPROVED** [review]

**Comment by elzody** [comment]
@juliusknorr Yep, only back to stable30. I already have a PR open for it: https://github.com/nextcloud/richdocuments/pull/4028

I just need to replicate what you added here and get it cleaned up

**Pull request merged**